src/libinjector/linux/linux_debug.c

```diff
@@ -0,0 +1,74 @@
+#include "linux_debug.h"
+
+void print_stack(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    PRINT_DEBUG("\nRSP: %lx\n", info->regs->rsp);
+    PRINT_DEBUG("Stack");
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+    uint32_t offset = 0;
+    for(int i=0; i < 16; i++)
+    {
+        ACCESS_CONTEXT(ctx,
+                       .translate_mechanism = VMI_TM_PROCESS_PID,
+                       .pid = info->proc_data.pid,
+                       .addr = (info->regs->rsp - offset + i*8)
+                      );
+        addr_t val = 0;
+        vmi_read_64(vmi, &ctx, &val);
+        if((i%4)==0)
+            PRINT_DEBUG("\n%016lx:", info->regs->rsp - offset + (i/4)*32);
+        PRINT_DEBUG(" %016lx", val);
+    }
+    PRINT_DEBUG("\n\n");
+    drakvuf_release_vmi(drakvuf);
+}
+
+static char *repeatStr (const char *str, size_t count)
```

Comment:
This is some voodu. Can we print registers in a normal way with "%0x"?

Reply:
copy paste `%s:\t%016lx\n` 16 times manually on a string?

Reply:
in my opinion, the debug codes need not be very good if they do the work and this seems to do the work, if you have any snippet related to repeating string multiple times in a better way please share it, i will change this function definition

Reply:
Ok, just make sure such functions copied from [stackoverflow](https://stackoverflow.com/questions/5770940/how-repeat-a-string-in-language-c) don't have any memory issues ;)